File: Leedcode Easy/Array/1. Two Sum/Solution.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        for(int i = 0; i < nums.length; i++){
            for (int j = i+1; j < nums.length; j++){
                int complement = target - nums[i];

                if (nums[j] == complement){
                    return new int[] {i, j};
                }
            }
        }
        throw new IllegalArgumentException("Invalid solution");
    }

    public static void main(String[] args){
        Solution solution = new Solution();
        int[] nums = { 2 , 7 , 11 , 15};
        int target = 9;
        int[] result = solution.twoSum(nums, target);

        System.out.println("Indices: " + result[0] + ',' + result[1]);
    }
}

File: Leedcode Easy/Array/1. Two Sum/Solution.rs
fn impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut sum_array = Vec::new();
        for i in 0..nums.len(){
            for j in i..nums.len(){
                if i != j && nums[i] + nums[j] == target {
                    sum_array.push(i as i32);
                    sum_array.push(j as i32);
                    return sum_array.to_vec();
                }
            }
        }
        Vec::new()
    }
    

    pub fn main() {
        let result = two_sum([2,4,6,10,14], 20);
        println!("Sum: {}", result);
    }
}